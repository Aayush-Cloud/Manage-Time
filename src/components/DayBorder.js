// @flow
import * as React from 'react';
import { Text, View } from 'react-native';
import ElevatedView from 'react-native-elevated-view';

import HorizontalDividerList from './HorizontalDividerList';
import styles from './styles/DayBorder.styles';

type Props = {
  date: string,
  children?: React.Node,
};

export default function DayBorder({ date, children }: Props) {
  return (
    <View>
      <View style={styles.sectionTitle}>
        <Text style={styles.dateText}>{date}</Text>
      </View>
      <ElevatedView elevation={2} style={styles.appointmentsContainer}>
        <HorizontalDividerList dividerStyle={styles.dividerStyle}>
          {children}
        </HorizontalDividerList>
      </ElevatedView>
    </View>
  );
}

DayBorder.defaultProps = {
  children: undefined,
};
