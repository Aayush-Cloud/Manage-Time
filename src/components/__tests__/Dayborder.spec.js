// @flow
import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import DayBorder from '../DayBorder';

describe('DayBorder component', () => {
  it('renders a given date text and all children beneath that', () => {
    const tree = renderer.create(
      <DayBorder date="some date">
        <Text>Child 1</Text>
        <Text>Child 2</Text>
      </DayBorder>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
