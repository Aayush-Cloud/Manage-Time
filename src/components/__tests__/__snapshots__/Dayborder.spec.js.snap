// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayBorder component renders a given date text and all children beneath that 1`] = `
<View>
  <View
    style={
      Object {
        "height": 48,
        "justifyContent": "center",
        "marginLeft": 16,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "Helvetica Neue",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      some date
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "shadowOffset": Object {
            "height": 1.2,
          },
          "shadowOpacity": 0.183,
          "shadowRadius": 1.08,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Child 1
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 1,
              "marginHorizontal": 0,
            },
            Object {
              "height": 0.5,
              "marginHorizontal": 8,
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Child 2
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 1,
              "marginHorizontal": 0,
            },
            Object {
              "height": 0.5,
              "marginHorizontal": 8,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
